addons/project_sale_expense/tests/test_project_profitability.py

```diff
@@ -199,7 +199,7 @@ def test_project_profitability(self):
             {'id': 'expenses', 'sequence': expense_sequence, 'billed': expense.currency_id.round(billed), 'to_bill': 0.0},
         )
 
-        expense_sheet._do_refuse('Test Cancel Expense')
```

Comment:
I see that currently `expense_sheet.approval_state == 'approve' `.
So it is not possible to call the corresponding action from the UI (without customizations) I think.
But there are currently other tests where we use `_do_refuse` on approved expense sheets.
See e.g.
https://github.com/odoo/odoo/blob/228927917e332295df53dcabb2bdb8e59aede092/addons/hr_expense/tests/test_expenses.py#L1523
Maybe `_do_refuse` should only be possible on sheets that are in `'submitted'` state?

Reply:
You actually can, button is moved further right though